WalletWasabi/Bases/ExceptionInfo.cs

```diff
@@ -4,15 +4,26 @@ namespace WalletWasabi.Bases
 {
 	public class ExceptionInfo
 	{
-		public ExceptionInfo(Exception exception)
+		private ExceptionInfo(Exception exception, DateTimeOffset firstAppeared, long count)
 		{
 			Exception = exception;
-			ExceptionCount = 1;
-			FirstAppeared = DateTimeOffset.UtcNow;
+			ExceptionCount = count;
+			FirstAppeared = firstAppeared;
+		}
+
+		public ExceptionInfo()
+			: this (new Exception(), DateTimeOffset.UtcNow, 0)
+		{
 		}
 
 		public Exception Exception { get; }
-		public long ExceptionCount { get; set; }
+		public long ExceptionCount { get; }
 		public DateTimeOffset FirstAppeared { get; }
+
+		public ExceptionInfo Is(Exception exception) =>
+			new ExceptionInfo(exception, DateTimeOffset.UtcNow, 1);
+
+		public ExceptionInfo Again() =>
```

Comment:
`Again` is fine. Still consider please other options like `Repeat`, `NextOccurrence`, etc.